src/Appwrite/Auth/OAuth2/Dailymotion.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+// Reference Material
+// https://developers.dailymotion.com/api/#authentication
+
+class Dailymotion extends OAuth2
+{
+    /**
+     * @var string
+     */
+    private string $endpoint = 'https://api.dailymotion.com';
```

Comment:
This endpoints for all the rest of the functions (tokens, refresh tokens, get user data etc.)